analyses/doublet-detection/scripts/01b_run-scrublet.py

```diff
@@ -58,23 +58,27 @@ def main() -> None:
     args = parser.parse_args()
 
     # Define and check files, directories
-    args.results_dir.mkdir(parents = True, exist_ok = True)
-
-    input_anndata_file = Path(args.input_anndata_file)
-    if not input_anndata_file.exists():
+    if not args.input_anndata_file.exists():
         print(
             "The input AnnData file could not be found at:",
             args.input_anndata_file,
             file=sys.stderr
         )
         sys.exit(1)
-    result_tsv = args.results_dir / (input_anndata_file.name.replace(".h5ad", "_scrublet.tsv"))
+
+    if not args.output_tsv_file.name.endswith(".tsv"):
+        print(
+            "The output TSV file must end in `.tsv`.",
+            file=sys.stderr
+        )
+        sys.exit(1)
```

Comment:
Minor, but if there are a series of argument checks, I tend to prefer to set a `arg_error=False` at the start of the checks and then use `arg_error=True` (or set to zero and increment, depending on mood) rather than immediately exiting. Then at the end of the checks: 
```
if arg_error:
  sys.exit(1)
```

I'd rather just print all the errors at once if I can.